Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -88,15 +89,33 @@ class ChunkedFileUploader {
             do {
                 // It's fine if it's already open, that's handled by ignoring the call
                 try file.openFile(fileURL: uploadInfo.videoFile)
+                let fileSize = file.fileSize
                 try file.seekTo(byte: lastByte)
                 let result = try await makeWorker().performUpload()
                 file.close()
-                
+
                 let success = UploadResult(
                     finalProgress: result.progress,
                     startTime: result.startTime,
                     finishTime: result.updateTime
                 )
+
+                let asset = AVAsset(url: uploadInfo.videoFile)
+
+                var duration: CMTime
+                if #available(iOS 16, *) {
+                    duration = try await asset.load(.duration)
+                } else {
+                    duration = asset.duration
+                }
+
+                Reporter.sharedInstance.report(
```

Comment:
What do you think about making `Reporter` a property of `ChunkedFileUploader`? This can help with automated tests for this down the road.

Reply:
Fine by me! I made it a shared instance because I was adapting this code example of handling redirects https://gist.github.com/Tulakshana/48d473a8aefa2d6b5f817f15870ae64f

I'm fine with this being a property of `ChunkedFileUploader`

Reply:
Ok, this is no longer a sharedInstance.